packages/fiori/src/BarcodeScannerDialog.ts

```diff
@@ -124,21 +134,35 @@ class BarcodeScannerDialog extends UI5Element {
 	loading!: boolean;
 
 	_codeReader: InstanceType<typeof BrowserMultiFormatReader>;
+	_previousOpenState: boolean;
 	dialog?: Dialog;
 	static i18nBundle: I18nBundle;
 
 	constructor() {
 		super();
+		this._previousOpenState = false;
 		this._codeReader = new BrowserMultiFormatReader();
 	}
 
 	static async onDefine() {
 		BarcodeScannerDialog.i18nBundle = await getI18nBundle("@ui5/webcomponents-fiori");
 	}
 
+	onBeforeRendering() {
+		if (this._previousOpenState !== this.open) {
+			if (this.open) {
+				this.show();
+			} else {
```

Comment:
Delete this block and add this instead:

```
	get _open() {
		return this.open || undefined;
	}

	onInvalidation(changeInfo: ChangeInfo) {
		if (changeInfo.type === "property" && changeInfo.name === "open") {
			changeInfo.newValue ? this.show() : this.close();
		}
	}
```

Reply:
I really like your suggestion to use `onInvalidation()`. Nice catch!